source/common/upstream/subset_lb.cc

```diff
@@ -514,8 +515,17 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const HostVector& hosts_added,
     bool host_seen = predicate_added.count(host) == 1;
     if (host_seen || predicate(*host)) {
       hosts->emplace_back(host);
-      if (host->healthy()) {
+      switch (host->health()) {
+      case Host::Health::Healthy:
         healthy_hosts->emplace_back(host);
+        break;
+      case Host::Health::Degraded:
+        degraded_hosts->emplace_back(host);
+        break;
+      case Host::Health::Unhealthy:
+        break;
+      default:
```

Comment:
Does it not compile without default? Typically if you specify all cases it should be fine to remove?